IMPORT INTO: update document for `SHOW IMPORT` (#21610)
pingcap/docs: sql-statements/sql-statement-show-import-job.md

Reviewer: ```suggestion
The `SHOW IMPORT JOB` statement is used to show `IMPORT` jobs created in TiDB. 

It supports the following two forms:

- `SHOW IMPORT JOBS`
- `SHOW IMPORT JOB <job-id>`

## Required privileges

- `SHOW IMPORT JOBS`: by default, this statement only shows jobs created by the current user. To view all import jobs, you need the `SUPER` privilege.
- `SHOW IMPORT JOB <job-id>`: only the creator of an import job or users with the `SUPER` privilege can use this statement to view a specific job.
```

---
title: SHOW IMPORT JOB
summary: An overview of the usage of SHOW IMPORT JOB in TiDB.
---

# SHOW IMPORT JOB

The `SHOW IMPORT JOB` statement is used to show `IMPORT` jobs created in TiDB. By default, `SHOW IMPORT JOBS` shows jobs created by the current user. To view all import jobs, you need the `SUPER` privilege.

## Required privileges

- `SHOW [RAW] IMPORT JOBS`: if a user has the `SUPER` privilege, this statement shows all import jobs in TiDB. Otherwise, this statement only shows jobs created by the current user.
- `SHOW [RAW] IMPORT JOB <job-id>`: only the creator of an import job or users with the `SUPER` privilege can use this statement to view a specific job.

## Synopsis

```ebnf+diagram
ShowImportJobsStmt ::=

>     'SHOW' RawOpt? 'IMPORT' 'JOBS' ShowLikeOrWhereOpt?

Reviewer 2: since `RAW` is not implemented yet, maybe update this later

Author: Ohhh, I see, only the parser part was added 😂

```ebnf+diagram
ShowImportJobsStmt ::=
    'SHOW' 'IMPORT' 'JOBS' ShowLikeOrWhereOpt?

ShowImportJobStmt ::=
    'SHOW' 'IMPORT' 'JOB' JobID

ShowLikeOrWhereOpt ::=

>     'LIKE' SimpleExpr

Reviewer 2: how `like` works, compare by complete result string or just ID?
```
show import jobs like '%1';
```

Author: It only matches the first column `Job_ID` (compared as a string), not the whole row — same as the standard `SHOW ... LIKE` behavior. In `buildShow` the planner rewrites the `LIKE` pattern onto the first output column, so `show import jobs like '%1'` returns the jobs whose `Job_ID` ends with `1` (1, 11, 21, ...). `fetchShowImportJobs` itself doesn't filter; `LIKE`/`WHERE` are applied as a `Selection` on top of the show plan.

https://github.com/pingcap/tidb/blob/26a449413784a3100d0a3751ebe858920e073fa9/pkg/planner/core/planbuilder.go#L3667-L3675

Added a note about this in https://github.com/pingcap/docs/pull/21610/commits/2188a5b38470899e55ef4f762236051c0b44886e

|   'WHERE' Expression
```

The output fields of the `SHOW IMPORT JOB` statement are described as follows:

| Column           | Description             |
|------------------|-------------------------|
| Job_ID           | The ID of the task                  |

> | Group_Key        | The group key of the task                  |

Reviewer: ```suggestion
| Job_ID           | The ID of the job                  |
| Group_Key        | The group key of the job                  |
```

| Cur_Step_Total_Size       | The total size of the data to be processed in the current step         |
| Cur_Step_Progress_Pct       | The estimated completion percentage of the current step        |
| Cur_Step_Speed       | The current data processing speed         |
| Cur_Step_ETA       | The estimated time remaining for the current step to complete       |

## Filtering import jobs

Only `SHOW IMPORT JOBS` supports filtering import jobs with a `WHERE` clause. `SHOW IMPORT JOB <job-id>` does not support a `WHERE` clause.

The `WHERE` clause can reference the output fields of `SHOW IMPORT JOBS`, including `Job_ID`, `Group_Key`, `Data_Source`, `Target_Table`, `Table_ID`, `Phase`, `Status`, `Source_File_Size`, `Imported_Rows`, `Result_Message`, `Create_Time`, `Start_Time`, `End_Time`, `Created_By`, `Last_Update_Time`, `Cur_Step`, `Cur_Step_Processed_Size`, `Cur_Step_Total_Size`, `Cur_Step_Progress_Pct`, `Cur_Step_Speed`, and `Cur_Step_ETA`.

## Example

Show all import jobs that the current user can view:

```sql
SHOW IMPORT JOBS;
```

Show a specific import job by job ID:

```sql
SHOW IMPORT JOB 2;
```

The output is as follows:

> | Job_ID | Group_Key | Data_Source | Target_Table | Table_ID | Phase | Status | Source_File_Size | Imported_Rows | Result_Message | Create_Time | Start_Time | End_Time | Created_By | Last_Update_Time | Cur_Step | Cur_Step_Processed_Size | Cur_Step_Total_Size | Cur_Step_Progress_Pct | Cur_Step_Speed | Cur_Step_ETA |

Reviewer 2: classic kernel still the old way, we need to keep the old way, and put those new ones somewhere else for Premium
`SHOW IMPORT GROUPS` too

Author: Classic kernel actually uses this same output format. There's no classic-vs-nextgen branching for `SHOW IMPORT JOB(S)` or `SHOW IMPORT GROUP(S)`: the column schema is shared and used unconditionally, `FillOneImportJobInfo` fills the same columns, and neither `fetchShowImportGroups` nor the `GROUP_KEY` option is kernel-gated. These columns were added for all kernels back in pingcap/tidb#62283. On classic, `Group_Key` simply shows `NULL` when no group key was set.

- schema: https://github.com/pingcap/tidb/blob/26a449413784a3100d0a3751ebe858920e073fa9/pkg/planner/core/planbuilder.go#L4583-L4587
- fill: https://github.com/pingcap/tidb/blob/26a449413784a3100d0a3751ebe858920e073fa9/pkg/executor/show.go#L2492
- groups: https://github.com/pingcap/tidb/blob/26a449413784a3100d0a3751ebe858920e073fa9/pkg/executor/show.go#L2704

Reviewer 2: I mean it's not in latest 8.5 release. but this PR is for master, it should be fine

| Start_Time       | The time when the task is started                     |
| End_Time         | The time when the task is ended            |
| Created_By       | The name of the database user who creates the task         |

> | Last_Update_Time       | The time when the task is last updated         |

Reviewer: ```suggestion
| Last_Update_Time       | The time when the job was last updated         |
```

| Cur_Step       | The specific sequential processing step of this job         |
| Cur_Step_Processed_Size       | The amount of data that has been processed within the current step         |
| Cur_Step_Total_Size       | The total size of the data to be processed in the current step         |
| Cur_Step_Progress_Pct       | The estimated completion percentage of the current step        |
| Cur_Step_Speed       | The current data processing speed         |
| Cur_Step_ETA       | The estimated time remaining for the current step to complete       |

> ## Filtering import jobs

Reviewer: ```suggestion
## Filter import jobs
```
